doc/query.md

```diff
@@ -39,25 +39,30 @@ Secondly, these documents are in the datastore:
 ```java
 { "name": "mike", 
   "age": 12, 
-  "pet": {"species": "cat"} };
+  "pet": {"species": "cat"},
+  "comment": "Mike goes to middle school and likes reading books." };
 
 { "name": "mike", 
   "age": 34, 
-  "pet": {"species": "dog"} };
+  "pet": {"species": "dog"},
+  "comment": "Mike is a doctor and likes reading books." };
 
 { "name": "fred", 
   "age": 23, 
-  "pet": {"species": "cat"} };
+  "pet": {"species": "cat"},
+  "comment": "Fred works for a startup out of his home office." };
 ```
 
-### Creating indexes
+### Indexes
 
 In order to query documents, indexes need to be created over
 the fields to be queried against.
 
```

Comment:
I added a section that points out the differences between JSON and TEXT indexes.  Done in a24a921